### PR TITLE
BUGS-2506: Update API calls to wpvulndb.com

### PR DESCRIPTION
**Pull request by ataylorme**
Add new `getPluginVulnerability` method that returns the results of an API call to wpvulndb.com for the specified plugin slug.

Refactor `is_vulnerable` to call `getPluginVulnerability` per plugin as `https://wpvulndb.com/data/plugin_vulns.json` is now a `404`.

Addresses #89.

### CODE REVIEW

File: php/pantheon/checks/plugins.php [hunk 1]
```diff
@@ -69,35 +69,99 @@ public function run() {
...
+	 * @return array containing vulnerability info or false
+	 */
+	protected function getPluginVulnerability( $plugin_slug )
+    {
+		// Get the vulnerability API token from the platform
+		$wpvulndb_api_token = getenv('PANTHEON_WPVULNDB_API_TOKEN');
+
+		// Throw an exception if there is no token
+		if( false === $wpvulndb_api_token || empty( $wpvulndb_api_token ) ) {
+			throw new \Exception('No WP Vulnerability DB API Token. Please ensure the PANTHEON_WPVULNDB_API_TOKEN environment variable is set');
```
Comment (ataylorme) [inline php/pantheon/checks/plugins.php]: @greg-1-anderson I'm not sure if throwing an exception is what we want to do here. The alternative is to not complete the check and fail silently, which is not desired either.

Comment (greg-1-anderson) [reply php/pantheon/checks/plugins.php]: I am +1 on total catastrophic failure when there is no API token set.

Well, throwing an exception is probably sufficient. No need to call `exit` or erase the hard drive or anything.

File: php/pantheon/checks/plugins.php [hunk 2]
```diff
@@ -69,35 +69,99 @@ public function run() {
...
 
 	/**
-	* Checks the plugin slug against the vulnerability db
+	 * Checks the plugin slug against the vulnerability db
+	 * @param $plugin_slug string (required) string representing the plugin slug
+	 *
+	 * @return array containing vulnerability info or false
+	 */
+	protected function getPluginVulnerability( $plugin_slug )
+    {
```
Comment (greg-1-anderson) [inline php/pantheon/checks/plugins.php]: Just one example of a formatting issue

**Comment by ataylorme** [comment]
> Do you know why the Travis tests are failing?

I do not

> Could you fix the code style issues? Looks like maybe there is a mix of tabs and spaces. Some things do not line up correctly.

Updated. I was editing in vim inside Lando

**Comment by ataylorme** [comment]
@greg-1-anderson it looks like [this cron test](https://github.com/pantheon-systems/wp_launch_check/blob/master/features/cron.feature#L51) is timing out (the build fails after 10 mins with no output).

The last time the build passed [was about a year ago](https://travis-ci.org/pantheon-systems/wp_launch_check/builds/294996072)

**Comment by greg-1-anderson** [comment]
Try removing just that cron test and see if the rest of it passes, maybe?

**Comment by ataylorme** [comment]
@greg-1-anderson with that test removed it [now fails with the missing token exception](https://travis-ci.org/pantheon-systems/wp_launch_check/jobs/492464513#L973).

Think it is okay to add `PANTHEON_WPVULNDB_API_TOKEN` with the token as an environment variable in Travis?

**Comment by ataylorme** [comment]
@greg-1-anderson you were right about `PANTHEON_WPVULNDB_API_TOKEN` not being loaded in Behat. See [7c3148fda0abac093a3f05b8a31195671f1e93b4](https://github.com/pantheon-systems/wp_launch_check/pull/91/commits/7c3148fda0abac093a3f05b8a31195671f1e93b4) for the fix.

**Comment by ataylorme** [comment]
After the plugin vulnerability were back to green I dug into why the cron sanity check test was failing.

The _given a file_ step, which is only used by the cron sanity check test, attempts to use `wp-cli` to run `mkdir -p` on the desired file path before running `file_put_contents`.

The `mkdir -p` command is run through the [wp-cli `esc_cmd` method](https://github.com/wp-cli/wp-cli/blob/master/php/utils.php#L213:L227). However, The class name had a lowercase `u`, calling `\WP_CLI\utils\esc_cmd` instead of `\WP_CLI\Utils\esc_cmd`.

The tests are now passing with the proper class name `\WP_CLI\Utils` in the _given a file_ step

**Pull request merged**